kernel/src/main/java/org/cristalise/kernel/lifecycle/instance/Activity.java

```diff
@@ -464,11 +464,15 @@ public Next getNext() {
      */
     @Override
     public void reinit(int idLoop) throws InvalidDataException {
-        Vertex[] outVertices = getOutGraphables();
+        Logger.msg(8, "Activity.reinit(id:%s, idLoop:%d) - parent:%s act:%s", getID(), idLoop, getParent().getName(), getPath());
+
         setState(getStateMachine().getInitialState().getId());
+
+        Vertex[] outVertices = getOutGraphables();
+
+        //NOTE: strange condition, activity can have zero or one outVertex and its id cannot be the loopId
         if (outVertices.length > 0 && idLoop != getID()) {
-            WfVertex nextAct = (WfVertex) outVertices[0];
-            nextAct.reinit(idLoop);
+            ((WfVertex) outVertices[0]).reinit(idLoop);
```

Comment:
By removing the `nextAct` variable the code became harder to understand. At least for me `nextAct` has more meaning than `outVertices[0]`

Reply:
it was badly variable